AirCasting/Notes/NotesHandler.swift

```diff
@@ -0,0 +1,80 @@
+// Created by Lunar on 16/12/2021.
+//
+
+import Foundation
+
+protocol NotesHandler {
+    func addNote(noteText: String)
+    func deleteNote(note: Note)
+    func updateNote(note: Note, newText: String)
+    func getNotes(completion: @escaping ([Note]) -> Void)
+    func fetchSpecifiedNote(number: Int, completion: @escaping (Note) -> Void)
+}
+
+class NotesHandlerDefault: NotesHandler {
+    var measurementStreamStorage: MeasurementStreamStorage
+    var sessionUUID: SessionUUID
+    var locationTracker: LocationTracker
+    
+    init(measurementStreamStorage: MeasurementStreamStorage, sessionUUID: SessionUUID, locationTracker: LocationTracker) {
+        self.measurementStreamStorage = measurementStreamStorage
+        self.sessionUUID = sessionUUID
+        self.locationTracker = locationTracker
+    }
+    
+    func addNote(noteText: String) {
+        measurementStreamStorage.accessStorage { [self] storage in
+            do {
+                let currentNumber = try storage.getNotes(for: sessionUUID).map(\.number).sorted(by: < ).last
+                try storage.addNote(Note(date: Date(),
+                                         text: noteText,
+                                         lat: locationTracker.googleLocation.last?.location.latitude ?? 20.0,
+                                         long: locationTracker.googleLocation.last?.location.longitude ?? 20.0,
+                                         number: (currentNumber ?? -1) + 1),
```

Comment:
gonna run into an Int overflow problem when user adds more than 9,223,372,036,854,775,807 notes 🤓🤓🤓

Reply:
🤫😬 Sounds not fun at all, but I believe in the users a bit in this scenario